pytest_fixtures/component/discovery.py

```diff
@@ -0,0 +1,46 @@
+from fauxfactory import gen_string
+import pytest
+
+
+@pytest.fixture(scope='module')
+def module_discovery_hostgroup(module_org, module_location, module_target_sat):
+    host = module_target_sat.api.Host(organization=module_org, location=module_location)
+    host.create_missing()
+    return module_target_sat.api.HostGroup(
+        organization=[module_org],
+        location=[module_location],
+        medium=host.medium,
+        root_pass=gen_string('alpha'),
+        operatingsystem=host.operatingsystem,
+        ptable=host.ptable,
+        domain=host.domain,
+        architecture=host.architecture,
+    ).create()
+
+
+@pytest.fixture(scope='module')
+def discovery_org(module_org, module_target_sat):
+    # Update default discovered host organization
+    discovery_org = module_target_sat.api.Setting().search(
+        query={'search': 'name="discovery_organization"'}
+    )[0]
+    default_discovery_org = discovery_org.value
+    discovery_org.value = module_org.name
+    discovery_org.update(['value'])
+    yield module_org
+    discovery_org.value = default_discovery_org
+    discovery_org.update(['value'])
+
+
+@pytest.fixture(scope='module')
+def discovery_location(module_location, module_target_sat):
+    # Update default discovered host location
+    discovery_loc = module_target_sat.api.Setting().search(
+        query={'search': 'name="discovery_location"'}
+    )[0]
+    default_discovery_loc = discovery_loc.value
+    discovery_loc.value = module_location.name
+    discovery_loc.update(['value'])
+    yield module_location
+    discovery_loc.value = default_discovery_loc
+    discovery_loc.update(['value'])
```

Comment:
Can't we combine these two fixtures into one? Since the main logic is to update org and loc we can keep that only  and  yielding `module_location` and `module_org` we can easily inherit these fixtures directly into the test

Reply:
ah, good point, we already have module_discovery_env in UI module, we could yield Box object for org and loc there and it can be moved to global fixtures, @adarshdubey-star wdyt?

Reply:
Good catch guys , I am trying your suggestions

Reply:
I will be addressing this in the UI refactor discussed with @Gauravtalreja1 , not increasing the scope of this PR @shubhamsg199